presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -55,8 +55,11 @@
     public static final String TIME_BOUNDARY_FILTER_TEMPLATE = "__TIME_BOUNDARY_FILTER_TEMPLATE__";
     public static final String TABLE_NAME_SUFFIX_TEMPLATE = "__TABLE_NAME_SUFFIX_TEMPLATE__";
     // Fields defining the query
-    // order map that maps the column definition in terms of input relation column(s)
-    private final LinkedHashMap<VariableReferenceExpression, Selection> selections;
+    // A map that maps the column definition in terms of input relation column(s)
+    private final Map<VariableReferenceExpression, Selection> selections;
+    // Outputs are the fields defined in Presto for what's expected from Pinot query.
+    // Outputs/groupByColumns/topNColumnOrderingMap should be part of the keys of selections.
+    private final LinkedHashSet<VariableReferenceExpression> outputs;
```

Comment:
could outputs be Set? LinkedHashSet is an implementation

Reply:
I would very much like to keep it as a LinkedHashSet ... I can't tell you the number of times this has bit me: If order is needed then it should be statically checked.